dotnet-testing/MockingWithNSubstitute/MockingWithNSubstitute/InvalidEmailException.cs

```diff
@@ -0,0 +1,5 @@
+﻿namespace MockingWithNSubstitute;
+
+public class InvalidEmailException : Exception
```

Comment:
You should go for three constructors here to keep it in line with the recommendation from Microsoft: https://learn.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions

public class InvalidEmailException : Exception
{
    public InvalidEmailException()
    {
    }

    public InvalidEmailException(string message)
        : base(message)
    {
    }

    public InvalidEmailException(string message, Exception inner)
        : base(message, inner)
    {
    }
}

Reply:
This is updated.